test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/Internal/ChunkVisitorTestFactory.cs

```diff
@@ -0,0 +1,144 @@
+            var tagHelperChunk = new TagHelperChunk(
+                name.ToLower(),
+                TagMode.SelfClosing,
+                new List<TagHelperAttributeTracker>(),
+                new List<TagHelperDescriptor>
+                {
+                    new TagHelperDescriptor
+                    {
+                        AssemblyName = $"{name}Assembly",
+                        TagName = name.ToLower(),
+                        TypeName = $"{name}Type",
+                    }
+                });
+
+            return tagHelperChunk;
+        }
+
+        private static ParentChunk GetNestedViewComponentTagHelperChunk(string name, bool visitedTagHelperChunks)
+        {
+            var parentChunk = new ParentChunk();
+            var tagHelperChunk = GetViewComponentTagHelperChunk(name, visitedTagHelperChunks);
+
+            if (visitedTagHelperChunks)
+            {
+                var tagHelperDescriptor = tagHelperChunk.Descriptors.First();
+                tagHelperDescriptor.TypeName = $"{_testNamespace}.{_testClass}.{tagHelperDescriptor.TypeName}";
+            }
+
+            parentChunk.Children.Add(tagHelperChunk);
+            return parentChunk;
+        }
+
+        private static TagHelperChunk GetViewComponentTagHelperChunk(string name, bool visitedTagHelperChunks)
+        {
+            // Add a view component tag helper.
+            var typeName = visitedTagHelperChunks ? $"{_testNamespace}.{_testClass}.{name}Type" : $"{name}Type";
+
+            var attribute = new TagHelperAttributeDescriptor
+            {
+                Name = "attribute",
+                PropertyName = "Attribute",
+                TypeName = "string"
+            };
+
+            var requiredAttribute = new TagHelperRequiredAttributeDescriptor
+            {
+                Name = "Attribute"
+            };
+
+            var tagHelperDescriptor = new TagHelperDescriptor
+            {
+                AssemblyName = $"{name}Assembly",
+                TagName = name.ToLower(),
+                TypeName = typeName,
+                Attributes = new List<TagHelperAttributeDescriptor>
+                {
+                    attribute
+                },
+                RequiredAttributes = new List<TagHelperRequiredAttributeDescriptor>
+                {
+                    requiredAttribute
+                }
+            };
+
+            tagHelperDescriptor.PropertyBag.Add(
+                ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey,
+                name);
+
+            var tagHelperChunk = new TagHelperChunk(
+                $"vc:{name.ToLower()}",
+                TagMode.SelfClosing,
+                new List<TagHelperAttributeTracker>(),
+                new List<TagHelperDescriptor>
+                {
+                    tagHelperDescriptor
+                });
+
+            return tagHelperChunk;
+        }
+
+        public static CodeGeneratorContext CreateDummyCodeGeneratorContext()
```

Comment:
http://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods